doc/docs/security/selinux.md

```diff
@@ -42,3 +43,55 @@ semanage permissive -a bluechi_t
 # add the permissive property to bluechi-agent
 semanage permissive -a bluechi_agent_t
 ```
+
+## Allowing access to restricted units
+
+The SELinux policy of BlueChi allows it to manage all systemd units.
+
+However, when installing some applications and their respective systemd units, e.g. via `dnf`, there might also be additional SELinux Policies installed which prevent BlueChi from managing these. For example, when installing `httpd` on Fedora, it installs also the systemd unit `httpd.service` and the [policy module for apache](https://github.com/fedora-selinux/selinux-policy/blob/rawhide/policy/modules/contrib/apache.te). When trying to start the service via `bluechictl`, the policy will prevent certain operations:
+
+```bash
+# The apache policy prevents BlueChi from managing the httpd service
+$ bluechictl stop <node-name> httpd.service
+Failed to issue method call: SELinux policy denies access: Permission denied
+
+# However, using systemctl works as this is enabled by default in the apache policy
+$ systemctl stop httpd.service
+```
+
+In order to allow BlueChi to manage the `httpd.service`, the necessary allow rule(s) need to be added. The [audit2allow](https://man7.org/linux/man-pages/man1/audit2allow.1.html) tool can be used to generate these rules based on AVCs.
+
+First, run all operations with BlueChi that should be allowed. These will fail and create the AVCs used by `audit2allow`. Then use `audit2allow` to generate the allow rules and create the policy package (.pp) which can be installed via `semodule`. The following snippet shows an example for the httpd.service:
+
+```bash
+# generate AVC for the status operation
+$ bluechictl status <node-name> httpd.service
+Failed to issue method call: SELinux policy denies access: Permission denied
+
+# generate AVC for the stop operation
+$ bluechictl stop <node-name> httpd.service
+Failed to issue method call: SELinux policy denies access: Permission denied
+
+# view the type enforcement rule that allows the denied operations
+$ audit2allow -a
+#============= bluechi_agent_t ==============
+
+allow bluechi_agent_t httpd_unit_file_t:service { status stop };
+
+# create the policy package (.pp) and type enforcement file (.te)
+$ audit2allow -a -M httpd-allow
+******************** IMPORTANT ***********************
+To make this policy package active, execute:
+
+semodule -i httpd-allow.pp
+
+$ ls
+httpd-allow.pp httpd-allow.te
+
+# install policy package and run the previously prevented operation
+$ semodule -i httpd-allow.pp
+$ bluechictl status <node-name> httpd.service
+UNIT		    | LOADED	| ACTIVE	| SUBSTATE	| FREEZERSTATE	| ENABLED	|
+---------------------------------------------------------------------------------
+httpd.service	| loaded	| inactive	| dead		| running	    | disabled	|
+```
```

Comment:
(If possible, not a must) Could you please also include the output when the new selinux rules are loaded? Example, bluechi agent is able to load everything and bluechictl is working out of the box?

Reply:
You mean the output of the `semodule -i <name>.pp`? When I ran it there was no output. A subsequent `bluechictl status` just worked afterwards. 
When adding the `-v` option I get:
```bash
$ semodule -v -i httpd-allow.pp
Attempting to install module 'httpd-allow.pp':
Ok: return value of 0.
Committing changes:
Ok: transaction number 28.
```
Not sure if this information is useful, though?!

Reply:
I meant, would be nice to see that after your rule is loaded, bluechictl is back working out of box.  Sorry, maybe I was not clear enough.

Reply:
Ah ok. That should already be covered by the last two statements in the snippet, though.